test/CMakeLists.txt

```diff
@@ -12,6 +12,11 @@ rock_testsuite(unit_test_filter
     #DEPS_PKGCONFIG base-types
     DEPS numeric
 )
+rock_testsuite(unit_test_savitzkygolayfilter
```

Comment:
I would suggest to use the following pattern instead of introducing another test binary
```
BOOST_AUTO_TEST_SUITE(savitzky_golay)

BOOST_AUTO_TEST_CASE(5pt_smooth_quadratic)
{...}

BOOST_AUTO_TEST_SUITE_END()
```
Allows you to run the full test suite with:
<test-binary> --run_test=savitzky_golay
or individual tests as
<test-binary> --run_test=savitzky_golay/5pt_smooth_quadratic